tests/rptest/scale_tests/tiered_storage_stability_test.py

```diff
@@ -209,3 +212,85 @@ def stage_decommission(self):
+            # with an inordinate number of local segments.
+            'retention.local.target.bytes': 2 * self.scaled_segment_size,
+            'cleanup.policy': 'delete',
+            'partition_autobalancing_node_availability_timeout_sec': self.unavailable_timeout,
+            'partition_autobalancing_mode': 'continuous',
+            'raft_learner_recovery_rate': 10 * 1024 * 1024 * 1024,
+        }
+        self.rpk.create_topic(self.topic_name,
+                              partitions=self.scaled_num_partitions,
+                              replicas=3,
+                              config=config)
+
+        try:
+            producer = KgoVerifierProducer(
+                self.test_context,
+                self.redpanda,
+                self.topic_name,
+                msg_size=128 * 1024,
+                msg_count=5 * 1024 * 1024 * 1024 * 1024,
+                rate_limit_bps=self.scaled_data_bps,
+                custom_node=[self.preallocated_nodes[0]])
+            producer.start()
+            wait_until(lambda: producer.produce_status.acked > 10000,
+                       timeout_sec=60,
+                       backoff_sec=1.0)
+
+            # S3 up -> down -> up
+            self.stage_block_s3()
+
+        finally:
+            producer.stop()
+            producer.wait(timeout_sec=600)
+            self.free_preallocated_nodes()
+
+    def _cloud_storage_no_new_errors(self, redpanda, logger=None):
+        num_errors = redpanda.metric_sum(
+            "redpanda_cloud_storage_errors_total",
+            metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS)
+        increase = (num_errors - self.last_num_errors) if self.last_num_errors>0 else 0
+        self.last_num_errors = num_errors
+        if logger:
+            logger.info(
+                f"Cluster metrics report {increase} new cloud storage errors ({num_errors} overall)"
+            )
+        return increase == 0
+
+    def stage_block_s3(self):
+        self.logger.info(f"Getting the first 100 segments into the cloud")
+        wait_until(lambda: nodes_report_cloud_segments(self.redpanda, 100, self
+                                                       .logger),
+                   timeout_sec=120,
+                   backoff_sec=5)
+        self.logger.info(f"Blocking S3 traffic for all nodes")
+        self.last_num_errors = 0
+        with firewall_blocked(self.redpanda.nodes, self.s3_port):
+            # wait for the first cloud related failure + one minute
+            wait_until(lambda: not self._cloud_storage_no_new_errors(self.redpanda, self.logger),
+                       timeout_sec=600, backoff_sec=10)
+            time.sleep(60)
+        # make sure nothing is crashed
+        wait_until(self.redpanda.healthy, timeout_sec=60, backoff_sec=1)
+        self.logger.info(f"Waiting for S3 errors to cease")
+        wait_until(lambda: self._cloud_storage_no_new_errors(self.redpanda, self.logger),
+                       timeout_sec=600, backoff_sec=20)
```

Comment:
I think this is okay for now; we'll probably want to update this to ensure the number of segments increases once S3 is unblocked.

Reply:
agree